mmv1/products/dlp/DiscoveryConfig.yaml

```diff
@@ -485,6 +491,128 @@ properties:
             # The fields below are necessary to include the "secretsDiscoveryTarget" target in the payload
           send_empty_value: true
           allow_empty_object: true
+        - !ruby/object:Api::Type::NestedObject
+          name: cloudStorageTarget
+          description: Cloud Storage target for Discovery. The first target to match a bucket will be the one applied.
+          properties:
+            - !ruby/object:Api::Type::NestedObject
+              name: filter
+              description: The buckets the generation_cadence applies to. The first target with a matching filter will be the one to apply to a bucket.
+              required: true
+              properties:
+                - !ruby/object:Api::Type::NestedObject
+                  name: collection
```

Comment:
For some of these nested fields, you may want to make them required, specifically if there is only one field that can be used. This can help prevent users from accidentally using invalid config like an empty `{}`, but it also usually works best when you do not expect more fields to be added in the future.

Reply:
The filter is required — the nested fields are not required though. Our API-side validation will check for an invalid filter, but we're hesitant to add more restrictions on the terraform side because we might add more possible fields to the filter.